geode-core/src/test/java/org/apache/geode/internal/cache/xmlcache/CacheXmlParserJUnitTest.java

```diff
@@ -128,9 +128,16 @@ public void testCacheXmlParserWithSimplePool() {
    */
   @Test
   public void testCacheXmlParserWithSimplePoolXerces() {
-    System.setProperty("javax.xml.parsers.SAXParserFactory",
+    String prevParserFactory = System.setProperty("javax.xml.parsers.SAXParserFactory",
```

Comment:
You can make this cleanup of the system properties more robust by using the `RestoreSystemProperties` JUnit rule.  All you need to do is add this as a member variable of the test class: 
```
  @Rule
  public final RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
```

(For an example see [LocatorLauncherTest](https://github.com/apache/geode/blob/d1db2f02d2ce45a437b34488934e5b1d53c7b5ca/geode-core/src/test/java/org/apache/geode/distributed/LocatorLauncherTest.java).)